docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,141 @@
+
+**Pointer types**
+
+This roadmap will refer to trivial value types that hold pointers as “pointer types.” These types include pointers themselves and types which are composed of any pointer types (potentially including other types as well). The pointers held by pointer types refer to memory that is *not owned* by the pointer type (making pointer types a “view” or “projection” into memory). While pointer types are very similar to trivial types with respect to their lifetime operations and the fact that they map similarly in these four language, they differ in the fact that while they themselves are not inherently unsafe, they may be used in unsafe APIs (discussed later).
+
+#### Owned types
+
+This category of types refers to types that own memory. These types are pointer types where the type *does own* the memory that their pointers point to. C++ often uses copy constructors and destructors to manage the lifetime of owned types. Therefore, the Swift compiler should assume that pointer types with custom copy constructor and destructors own their memory (and should provide annotations to override this assumption).
+
+#### Lifetime and safety of owned types
+
+Value types that own memory do not natively exist in Swift today. Because Swift was not built around this kind of value type, Swift handles the lifetime of value types in a subtly different way from C++. This subtle difference makes a naive mapping of C++ value types that own memory to native Swift value types extremely dangerous, especially when dealing with projections of owned storage. Let’s look at an example Swift program that naively imports some owned type and returns a projections of it: 
+
+```
+var v = vector(1) 
+let start = v.begin() 
+doSomething(start) 
+fixLifetime(v)
+```
+
+Here, because Swift inserts a copy before the call to begin, v projects a dangling reference. This is an example of how subtly different lifetime models make using C++ types from Swift hard, if their semantics aren’t understood by the compiler. 
+
+To make these APIs safe and usable, Swift cannot import unsafe projections of types that own memory, because they don’t fit the Swift model. Instead, the Swift compiler can try to infer what, semantically, the API is trying to do, or the library author can provide this information via annotations. In this case, the Swift compiler can infer that begin returns an iterator, which Swift can represent through the existing, safe Swift iterator interface. In the future, Swift may gain the necessary ownership features to import and use projections of types that own memory in a safe way, but today, this is not possible, and importing these projections would result in a *less safe* API in Swift than in C++ (violating the first goal above). Other than methods which project owned memory, owned types, their instances, methods on owned types, and other APIs that use owned types are generally considered to be safe and usable.
```

Comment:
```suggestion
To make these APIs safe and usable, Swift cannot import unsafe projections of types that own memory, because they don’t fit the Swift model. Instead, the Swift compiler can try to infer what, semantically, the API is trying to do, or the library author can provide this information via annotations. In this case, the Swift compiler can infer that `begin` returns an iterator, which Swift can represent through the existing, safe Swift iterator interface. In the future, Swift may gain the necessary ownership features to import and use projections of types that own memory in a safe way, but today, this is not possible, and importing these projections would result in a *less safe* API in Swift than in C++ (violating the first goal above). Other than methods which project owned memory, owned types, their instances, methods on owned types, and other APIs that use owned types are generally considered to be safe and usable.
```